### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by bwsw**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+resolver = "2"
+members = [
+    "media_gateway_client",
+    "media_gateway_common",
+    "media_gateway_server"
+]
+
+[workspace.dependencies]
+savant_core = { git = "https://github.com/insight-platform/savant-rs", tag = "0.2.30" }
+serde_json = "1"
```
Comment (bwsw) [inline Cargo.toml:12]: JSON is slow for data. If you want avoid protobuf, use the `bincode` crate.

File: Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+[workspace.package]
+version = "0.1.0"
+edition = "2021"
+authors = ["Ivan Kudriavtsev <ivan.a.kudryavtsev@gmail.com>", "Ksenia Vazhdaeva <vazhdaeva_ks@bw-sw.com"]
+description = "Media Gateway (client and server)"
+homepage = "https://github.com/insight-platform/MediaGateway"
+repository = "https://github.com/insight-platform/MediaGateway"
+readme = "README.md"
+license = "In-Sight License"
```
Comment (bwsw) [inline Cargo.toml]: BSL-1.1

File: media_gateway_server/src/server/api.rs [hunk 3]
```diff
@@ -0,0 +1,16 @@
+use std::sync::Mutex;
+
+use actix_web::{post, web, Responder};
+
+use media_gateway_common::model::Media;
+
+use crate::server::service::GatewayService;
+
```
Comment (bwsw) [inline media_gateway_server/src/server/api.rs:10]: add a health check endpoint as well, please.

File: media_gateway_server/src/server/api.rs [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+#[post("/")]
+async fn gateway(
+    service: web::Data<Mutex<GatewayService>>,
+    query: web::Json<Media>,
```
Comment (bwsw) [inline media_gateway_server/src/server/api.rs]: This is slow; use `bincode`, `rkyv` or `protobuf`, not JSON.

File: Cargo.toml (hunk 2 above)
Comment (ksenia-vazhdaeva) [reply Cargo.toml]: Fixed

File: Cargo.toml (hunk 1 above)
Comment (ksenia-vazhdaeva) [reply Cargo.toml:12]: Fixed

File: media_gateway_server/src/server/api.rs (hunk 4 above)
Comment (ksenia-vazhdaeva) [reply media_gateway_server/src/server/api.rs]: Fixed

File: media_gateway_server/src/server/api.rs (hunk 3 above)
Comment (ksenia-vazhdaeva) [reply media_gateway_server/src/server/api.rs:10]: Fixed

**Pull request merged**